### PR TITLE
Alias old annotations to fully support Lombok's `checkerframework = true` option

### PR DESCRIPTION
**Pull request by kelloggm**
I copied the delombok'd test code directly from Lombok's source, so this should match exactly what they do.

The downside is that we have to go back to releasing annotations in the `org.checkerframework.checker.builder.qual` package, but I don't think that's a deal breaker. I considered marking them as deprecated, but that caused extra warnings when processing delombok'd code so I decided against it.

The annotations this adds support for are:
* `@CalledMethods` in the `builder.qual` namespace, which behaves identically to `@CalledMethods`
* `@NotCalledMethods`, which they (speculatively) generate, does nothing
* `@ReturnsReceiver` is trusted if it's found but no `@This` annotation is. This is the only one I'm a little worried about, but I think it's probably fine. The Javadoc on it makes it clear that users should use `@This` instead, and that's also what our README etc say

### CODE REVIEW

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -150,7 +167,20 @@ boolean returnsThis(final MethodInvocationTree tree) {
     AnnotatedTypeMirror methodATm = rrATF.getAnnotatedType(methodEle);
     AnnotatedTypeMirror rrType =
         ((AnnotatedTypeMirror.AnnotatedExecutableType) methodATm).getReturnType();
-    return rrType != null && rrType.hasAnnotation(This.class);
+    if (rrType != null && rrType.hasAnnotation(This.class)) {
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java]: Just write this `if` condition as an or?

File: returnsrcvr-qual/src/main/java/org/checkerframework/checker/builder/qual/ReturnsReceiver.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This declaration annotation indicates that the method on which it is written returns exactly the
+ * receiver object.
+ *
+ * <p>This annotation can only be written on a method declaration. It is inherited by all overrides
+ * of that method.
```
Comment (msridhar) [inline returnsrcvr-qual/src/main/java/org/checkerframework/checker/builder/qual/ReturnsReceiver.java]: Does the new test case also test the override functionality?  Not sure if it matters

Comment (kelloggm) [reply returnsrcvr-qual/src/main/java/org/checkerframework/checker/builder/qual/ReturnsReceiver.java]: It does not. I also don't think it really matters, but I can add a test.

**Comment by kelloggm** [comment]
Wrt to the lombok thing - I tried to reproduce it with their actual release and couldn't. As far as I can tell, the issue is in their internal "verbose" delombok mode, which isn't accessible from their actual release. I can still report it, but it's kind of silly without a reproducible example since the only non-compilable files are their own tests.

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java (hunk 1 above)
Comment (kelloggm) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java]: Good thought. Done.

**Comment by msridhar** [comment]
Sorry I meant to get Lombok to generate `@This` and our other “modern”
annotations

**Comment by kelloggm** [comment]
Yeah, I can make a PR to that effect. I'll wait to submit it until we have a new OCC release, though.

**Pull request merged**